### PR TITLE
`RadProfile` line cleanup

### PR DESCRIPTION
**Pull request by nollety**
# Description

Resolves eradiate/eradiate-issues#143

# Checklist

- [x] The code follows the relevant coding guidelines
- [x] The code generates no new warnings
- [x] The code is appropriately documented
- [x] The code is tested to prove its function
- [x] The feature branch is rebased on the current state of the `main` branch
- [x] I give permission that the Eradiate project may redistribute my contributions under the terms of its license

### CODE REVIEW

**Comment by nollety** [comment]
On io, full test suite ran in 20 minutes

File: tests/02_eradiate/02_system/test_compare_rami4atm.py [hunk 1]
```diff
@@ -86,6 +86,7 @@ def test_compare_rami4atm_onedim(
         "zeniths": np.linspace(-90, 90, 19),
         "azimuths": 0.0,
         "spp": 1e4,
+        "spectral_cfg": ertsc.measure.MeasureSpectralConfig.new(bin_set="10nm", bins="550")
```
Comment (leroyvn) [inline tests/02_eradiate/02_system/test_compare_rami4atm.py:89]: Black pass please

File: tests/02_eradiate/02_system/test_compare_rami4atm.py [hunk 2]
```diff
@@ -216,6 +217,7 @@ def test_compare_rami4atm_rami(
...
```
Comment (leroyvn) [inline tests/02_eradiate/02_system/test_compare_rami4atm.py:220]:
```suggestion
        "spectral_cfg": ertsc.measure.MeasureSpectralConfig.new(wavelengths=[550])
```

File: tests/02_eradiate/02_system/test_compare_rami4atm.py [hunk 3]
```diff
@@ -145,7 +146,7 @@ def test_compare_rami4atm_onedim(
 @pytest.mark.parametrize("sza", [0.0, 30.0, 60.0])
 @pytest.mark.parametrize("lai", [1.0, 2.0, 3.0])
 def test_compare_rami4atm_rami(
-    mode_mono_double, sza, lai, artefact_dir, ert_seed_state
+    mode_ckd_double, sza, lai, artefact_dir, ert_seed_state
```
Comment (leroyvn) [inline tests/02_eradiate/02_system/test_compare_rami4atm.py:149]: Running this in CKD mode makes little sense IMO: the RAMI experiment is intended to be monochromatic.
```suggestion
    mode_mono_double, sza, lai, artefact_dir, ert_seed_state
```

File: tests/02_eradiate/02_system/test_onedim_atmosphere.py [hunk 4]
```diff
@@ -29,20 +29,35 @@ def test_heterogeneous_atmosphere_contains_particle_layer(
     ------------------
     The two experiments produce *exactly* the same results.
     """
+    # measure
+    measure = {
+        "type": "distant",
+        "id": "measure",
+        "construct": "from_viewing_angles",
+        "zeniths": np.linspace(-90, 90, 19),
```
Comment (leroyvn) [inline tests/02_eradiate/02_system/test_onedim_atmosphere.py:37]:
```suggestion
        "zeniths": np.arange(-90, 91, 15),
```

File: tests/02_eradiate/02_system/test_onedim_atmosphere.py [hunk 5]
```diff
@@ -73,9 +88,21 @@ def test_heterogeneous_atmosphere_contains_molecular_atmosphere(
...
```
Comment (leroyvn) [inline tests/02_eradiate/02_system/test_onedim_atmosphere.py:96]:
```suggestion
        "zeniths": np.arange(-90, 91, 15),
```

**Review by leroyvn — CHANGES_REQUESTED** [review]
Hi @nollety, thanks for this much needed cleanup pass. There are a few points to be addressed:

* You introduced a kernel submodule regression in your first commit, please fix that.
* The test workload can be optimised by reducing the number of angles and with a slightly more refined mode selection choice.
* See detailed comments for the rest.

While suggesting changes, I made a few experiments, which are committed to a branch on this repo. It notably fixes the submodule regression, you can start from it if you wish.

**Comment by nollety** [comment]
On io, full test suite now runs in ~14 minutes

**Comment by nollety** [comment]
Hi @leroyvn, thanks a lot for your feedback! Apologies for the submodule regression. I'll address your remaining comment and see if I can optimise further the test run time.

**Pull request merged**